From 9f6bb045a0790484ab80f95994d36e8c6ccf100d Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 24 Dec 2012 13:01:00 +0100
Subject: [PATCH 163/204] fstab-generator: properly detect bind mounts

This kinda undoes a83cbaccd03c3f28e47e9330f4a22ff65ce4b561 and
1d634e21b453f3c80d7c6c4bd90a6b84e42a3d2a but corrects the original code
to compare the mount type with "bind" rather than the mount options.
(cherry picked from commit f9ea108e7c3544c03822277a1112a48dc62f6ed4)

Conflicts:
	src/fstab-generator/fstab-generator.c
---
 src/fstab-generator/fstab-generator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fstab-generator/fstab-generator.c b/src/fstab-generator/fstab-generator.c
index be7eabe..23e5051 100644
--- a/src/fstab-generator/fstab-generator.c
+++ b/src/fstab-generator/fstab-generator.c
@@ -199,7 +199,7 @@ static bool mount_is_bind(struct mntent *me) {
 
         return
                 hasmntopt(me, "bind") ||
-                streq(me->mnt_opts, "bind");
+                streq(me->mnt_type, "bind");
 }
 
 static bool mount_is_network(struct mntent *me) {
-- 
1.8.1.5

